File: kanban-sticky-notes react app/src/services/KanbanService.js
import axios from 'axios'

const SERVER_URL = 'http://localhost:8080/';
const BOARD_REST_API_URL = SERVER_URL + 'api/boards';
const CATEGORY_REST_API_URL = SERVER_URL + 'api/categories';
const NOTE_REST_API_URL = SERVER_URL + 'api/notes';


// BOARD

function getBoards() {
    return axios.get(BOARD_REST_API_URL);
}

function getBoardById(id) {
    return axios.get(BOARD_REST_API_URL + `/${id}`);
}

// CATEGORY
function getCategoriesByBoardId(id) {
    return axios.get(CATEGORY_REST_API_URL, {
        params: {
            board: id
        }
    });
}

// NOTE
function getNotesByCategoryId(id) {
    return axios.get(NOTE_REST_API_URL, {
        params: {
            category: id
        }
    })
}

function getNoteById(id) {
    return axios.get(NOTE_REST_API_URL + `/${id}`);
}




export {getBoards, getBoardById, getCategoriesByBoardId, getNotesByCategoryId, getNoteById};

File: Kanban REST API/src/main/java/eo/kanbanboard/service/KanbanService.java
package eo.kanbanboard.service;

import java.util.List;

import eo.kanbanboard.entity.*;

public interface KanbanService {
	
	// Board
	public List<Board> getBoards();
	public Board getBoardById(int id);
	public Board addBoard(Board board);
	public Board updateBoard(Board board);
	public void deleteBoard(Board board);
	
	// Category
	public List<Category> getCategoriesByBoardId(int id);
	public Category getCategoryById(int id);
	public Category addCategory(Category category);
	public Category updateCategory(Category category);
	public void deleteCategory(Category category);
	
	// Note
	public List<Note> getNotesByCategoryId(int id);
	public Note getNoteById(int id);
	public Note addNote(Note note);
	public Note updateNote(Note note);
	public void deleteNote(Note note);
}
